serverless/jobs/how-to/reference-secret-in-job.mdx

```diff
@@ -0,0 +1,80 @@
+---
+meta:
+  title: How to reference secrets in Serverless Jobs
+  description: Steps to reference secrets from Secret Manager in your Serverless Jobs.
+content:
+  h1:  How to reference secrets in Serverless Jobs
+  paragraph: Steps to reference secrets from Secret Manager in your Serverless Jobs.
+tags: serverless jobs secrets secret-manager environment-variable
+dates:
+  validation: 2024-10-27
+  posted: 2024-10-27
+categories:
+  - serverless
+  - jobs
+---
+
+Serverless Jobs seamlessly integrates with [Secret Manager](/identity-and-access-management/secret-manager/), which allows you to store, manage, and access sensitive information, such as credentials, SSH keys, SSL/TLS certificates, or any key/value pairs you need to secure.
+
+You can reference any secret stored in Secret Manager in a job, without having to hardcode any sensitive data.
+
+A [job run](/serverless/jobs/concepts/#job-run) accesses each secret at startup, and each access generates a call to the Secret Manager API, which is billed accordingly. Refer to the [Secret Manager pricing](/identity-and-access-management/secret-manager) for more information.
+
+<Macro id="requirements" />
+
+- A Scaleway account logged into the [console](https://console.scaleway.com)
+- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
+- [Created a Serverless Job](/serverless/jobs/how-to/create-job-from-scaleway-registry/)
+- [Created a secret](/identity-and-access-management/secret-manager/how-to/create-secret/)
+
+## Reference a secret in a job
+
+1. Click **Jobs** in the **Serverless** section of the side menu. The jobs page displays.
+
+2. Click the name of the job to which you want to add a secret, then open the **Settings** tab.
+
+3. In the **Secrets references** section, click **+ Add secret reference**. A pop-up displays.
+
+4. Select the secret you want to reference, and the desired version, then click **Select reference method**.
+
+5. Select the desired reference method:
+
+  - **File**: copies the encrypted value of your secret to a file stored at the indicated location within your container. This method is recommended for large or complex data. For example, if your secret is a certificate, you can store it as a file in the `/my-certificates` folder in your container.
```

Comment:
```suggestion
  - **File**: copies the encrypted value of your secret to a file stored at the indicated location within your job. This method is recommended for large or complex data. For example, if your secret is a certificate, you can store it as a file in the `/my-certificates` folder in your job
```

I think, from a product perspective, the client doesn’t need to be aware of the underlying technologies we use.